From 455c4ae2c70348a5842835d2f67f7cd8e665a2a6 Mon Sep 17 00:00:00 2001
From: Tomi Valkeinen <tomi.valkeinen@ideasonboard.com>
Date: Thu, 21 Sep 2023 16:03:07 +0300
Subject: [PATCH] media: rp1: fe: Use ~0, not -1, when working with unsigned
 values

Use ~0, not -1, when working with unsigned values (-1 is not unsigned).

Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ideasonboard.com>
---
 drivers/media/platform/raspberrypi/rp1_cfe/pisp_fe.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/media/platform/raspberrypi/rp1_cfe/pisp_fe.c
+++ b/drivers/media/platform/raspberrypi/rp1_cfe/pisp_fe.c
@@ -372,7 +372,7 @@ void pisp_fe_submit_job(struct pisp_fe_d
 void pisp_fe_start(struct pisp_fe_device *fe)
 {
 	pisp_fe_reg_write(fe, FE_CONTROL, FE_CONTROL_RESET);
-	pisp_fe_reg_write(fe, FE_INT_STATUS, -1);
+	pisp_fe_reg_write(fe, FE_INT_STATUS, ~0);
 	pisp_fe_reg_write(fe, FE_INT_EN, FE_INT_EOF | FE_INT_SOF | FE_INT_LINES0 | FE_INT_LINES1);
 	fe->inframe_count = 0;
 }
@@ -383,7 +383,7 @@ void pisp_fe_stop(struct pisp_fe_device
 	pisp_fe_reg_write(fe, FE_CONTROL, FE_CONTROL_ABORT);
 	usleep_range(1000, 2000);
 	WARN_ON(pisp_fe_reg_read(fe, FE_STATUS));
-	pisp_fe_reg_write(fe, FE_INT_STATUS, -1);
+	pisp_fe_reg_write(fe, FE_INT_STATUS, ~0);
 }
 
 static struct pisp_fe_device *to_pisp_fe_device(struct v4l2_subdev *subdev)
